controllers/toolchainclusterhealth/healthchecker_test.go

```diff
@@ -0,0 +1,176 @@
+			tctype:            "unstable",
+			apiendpoint:       "http://unstable.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{unhealthy(), notOffline()},
+			status:            toolchainv1alpha1.ToolchainClusterStatus{},
+		},
+		"StableNoCondition": {
+			tctype:            "stable",
+			apiendpoint:       "http://cluster.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{healthy()},
+			status:            toolchainv1alpha1.ToolchainClusterStatus{},
+		},
+		"NotFoundNoCondition": {
+			tctype:            "not-found",
+			apiendpoint:       "http://not-found.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{offline()},
+			status:            toolchainv1alpha1.ToolchainClusterStatus{},
+		},
+		//ToolchainCluster.status already contains conditions
+		"UnstableContainsCondition": {
+			tctype:            "unstable",
+			apiendpoint:       "http://unstable.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{unhealthy(), notOffline()},
+			status:            withStatus(healthy()),
+		},
+		"StableContainsCondition": {
+			tctype:            "stable",
+			apiendpoint:       "http://cluster.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{healthy()},
+			status:            withStatus(healthy()),
+		},
+		"NotFoundContainsCondition": {
+			tctype:            "not-found",
+			apiendpoint:       "http://not-found.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{offline()},
+			status:            withStatus(healthy()),
+		},
+		//if the connection cannot be established at beginning, then it should be offline
+		"OfflineConnectionNotEstablished": {
+			tctype:            "failing",
+			apiendpoint:       "http://failing.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{offline()},
+			status:            toolchainv1alpha1.ToolchainClusterStatus{},
+		},
+		//if no zones nor region is retrieved, then keep the current
+		"NoZoneKeepCurrent": {
+			tctype:            "stable",
+			apiendpoint:       "http://cluster.com",
+			clusterconditions: []toolchainv1alpha1.ToolchainClusterCondition{healthy()},
+			status:            withStatus(offline()),
+		},
+	}
+	for k, tc := range tests {
+		t.Run(k, func(t *testing.T) {
+			tctype, sec := newToolchainCluster(tc.tctype, tcNs, tc.apiendpoint, tc.status)
+			cl := test.NewFakeClient(t, tctype, sec)
+			reset := setupCachedClusters(t, cl, tctype)
+			defer reset()
+			cachedtc, found := cluster.GetCachedToolchainCluster(tctype.Name)
+			require.True(t, found)
+			cacheclient, err := kubeclientset.NewForConfig(cachedtc.RestConfig)
+			require.NoError(t, err)
+			healthChecker := &HealthChecker{
+				localClusterClient:     cl,
+				remoteClusterClient:    cachedtc.Client,
+				remoteClusterClientset: cacheclient,
+				logger:                 logger,
+			}
+			// when
+			err = healthChecker.updateIndividualClusterStatus(context.TODO(), tctype)
```

Comment:
It seems to me that, uf we add a constructor for the HealthChecker and export the `updateIndividualClusterStatus`, we can then move this tests to the ad-hoc package `toolchainclusterhealth_test`. That would make the `toolchainclusterhealth` package a lot cleaner. WDYT?

Reply:
Will be doing it in follow up Prs